enterprise/core-edge/src/main/java/org/neo4j/coreedge/discovery/HazelcastClient.java

```diff
@@ -67,29 +100,14 @@ public ClusterTopology currentTopology()
 
             try
             {
-                clusterTopology = HazelcastClusterTopology.fromHazelcastInstance( hazelcastInstance, log );
+                return hazelcastAccessor.apply( hazelcastInstance );
             }
             catch ( HazelcastInstanceNotActiveException e )
             {
                 hazelcastInstance = null;
+                exception = e;
             }
         }
-
-        return clusterTopology;
-    }
-
-    @Override
-    public void stop() throws Throwable
-    {
-        if ( hazelcastInstance != null )
-        {
-            hazelcastInstance.shutdown();
-        }
-    }
-
-    @Override
-    public void registerEdgeServer( AdvertisedSocketAddress address )
-    {
-        hazelcastInstance.getSet( EDGE_SERVERS ).add( address.toString() );
+        throw exception;
```

Comment:
`exception` might be `null` if you get out the `while` loop immediately on the `break`. No?